src/bun.js/test/jest.zig

```diff
@@ -1390,6 +1428,11 @@ pub const TestRunnerTask = struct {
         processTestResult(this, this.globalThis, result, test_, test_id, describe);
     }
 
+    pub fn assert_asserts(this: *TestRunnerTask, assert_type: NeededAssertType, assert_count: u32) void {
+        this.needed_assert_type = assert_type;
```

Comment:
given a test like this, jest will fail because it expects 0 assertions, but bun will pass because `hasAssertions` is the last to mark `needed_assert_type`. Would need to look at the jest implementation, but I think we need to check both `hasAssertions` and `assertions` after a test finishes
```js
test("debugging", () => {
  expect(1).toBe(1);
  expect.assertions(0);
  expect.hasAssertions();
});
```

Reply:
Jest keeps the state for `hasAssertions` and `assertions` separate alongside the state for the assertion counts. Should I do something similar by creating `pub var`s in `expect.zig` as Bun does for `active_test_expectation_counter`?:

https://github.com/oven-sh/bun/blob/dcf6f244a4033f5793cf86ce8ea597610ea208aa/src/bun.js/test/expect.zig#L42